src/haxelib/api/GlobalScope.hx

```diff
@@ -185,6 +185,18 @@ class GlobalScope extends Scope {
 			);
 			addLine('-D ${info.name}=${info.version}');
 
+			// custom defines if defined
+			if (info.customDefines != null && info.customDefines != "") {
+				var path = Path.join([resolved.path, info.customDefines]);
+				addLine('--macro addDefinesDescriptionFile(\'$path\')');
+			}
+
+			// custom metadatas if defined
+			if (info.customMetas != null && info.customMetas != "") {
+				var path = Path.join([resolved.path, info.customMetas]);
+				addLine('--macro addMetadataDescriptionFile(\'$path\')');
+			}
+
```

Comment:
We can add a version check here to keep compatible with older versions of the compiler:
```haxe
// we'll have to change this to "4.3.0" after the release
if (resolveCompiler().version > Semver.ofString("4.3.0-rc.1")) { 
    ...
}
```
Otherwise we get:
```
(unknown) : Method addMetadataDescriptionFile not found on class haxe.macro.Compiler
```